src/imds.erl

```diff
@@ -40,7 +40,7 @@ public_hostname() ->
 get_session_token() ->
     case role_name() of
         {ok, RoleName} ->
-            Result = imds_tokens(["iam/security-credentials/", http_uri:encode(RoleName)]),
+            Result = imds_tokens(["iam/security-credentials/", httpd_util:encode_hex(RoleName)]),
```

Comment:
That function is not in the documentation.
http://erlang.org/doc/man/httpd_util.html

Also it uses `http_uri:encode` but it just puts a 
```
-compile({nowarn_deprecated_function, [{http_uri, encode, 1}]}).
```
in the module.

Reply:
I'm aware of that but my thinking is this:
> OTP team doesn't really have a replacement for `http_uri:encode/1`, even when they _deprecated_ it. So, whenever they fix `httpd_util:encode_hex/1`, we'll immediately find the fix that we need.
> If they remove it, we'll know and we'll be able to change the code that uses it in the same way they change their code.
> If they change its internal implementation, our code will still be working but… if we really want to, we'll be able to change this code to match their changes.

If you think it's better for us to keep using `http_uri:encode/1` and add the `-compile` flag ourselves, I can do that. But I think we're safe using `httpd_util:encode_hex/1` since we don't change OTP versions _easily_ nor _frequently_ and therefore it will stay with us until, at least, next year. And even more… since it's still there in OTP24.

Reply:
We can _probably_ make it work better using what's described/linked in https://github.com/erlang/otp/issues/4884#issuecomment-848779119 … but I'm not entirely sure if that change should be part of this PR or a different one.

Reply:
So, I followed @zerth's advice on that thread and it seems to work as expected.